packages/grzctl/src/grzctl/commands/db.py

```diff
@@ -429,17 +421,116 @@ def modify(ctx: click.Context, submission_id: str, key: str, value: str):
         if not submission:
             raise SubmissionNotFoundError(submission_id)
         _ = db_service.modify_submission(submission_id, key, value)
-        console.print(f"[green]Updated {key} of submission '{submission_id}'[/green]")
+        console_err.print(f"[green]Updated {key} of submission '{submission_id}'[/green]")
+    except SubmissionNotFoundError as e:
+        console_err.print(f"[red]Error: {e}[/red]")
+        console_err.print(f"You might need to add it first: grz-cli db submission add {submission_id}")
+        raise click.Abort() from e
+    except Exception as e:
+        console_err.print(f"[red]An unexpected error occurred: {e}[/red]")
+        traceback.print_exc()
+        raise click.ClickException(f"Failed to update submission state: {e}") from e
+
+
+def _diff_metadata(
+    submission: Submission, metadata: GrzSubmissionMetadata, ignore_fields: set[str]
+) -> list[tuple[str, Any, Any]]:
+    changes = []
+
+    simple_fields = {"tan_g", "submission_date", "submission_type", "submitter_id", "disease_type"}
+
+    for field in simple_fields - ignore_fields:
+        if field == "tan_g" and metadata.submission.tan_g == ["0"] * 64:
+            raise ValueError(
+                "Refusing to populate a seemingly-redacted TAN (all zeros). "
+                "Add 'tan_g' to --ignore-field or use 'grzctl db submission modify' directly."
+            )
+        submission_attr = getattr(submission, field)
+        metadata_attr = getattr(metadata.submission, field)
+        if submission_attr != metadata_attr:
+            changes.append((field, submission_attr, metadata_attr))
+
+    # pseudonym (TODO: change after phase 0)
+    if "pseudonym" not in ignore_fields and (submission.pseudonym != metadata.submission.local_case_id):
+        if not metadata.submission.local_case_id:
+            raise ValueError(
+                "Refusing to populate a seemingly-redacted local case ID (empty). "
+                "Add 'pseudonym' to --ignore-field or use 'grzctl db submission modify' directly."
+            )
+        changes.append(("pseudonym", submission.pseudonym, metadata.submission.local_case_id))
+
+    # data node id
+    if "data_node_id" not in ignore_fields and (submission.data_node_id != metadata.submission.genomic_data_center_id):
+        changes.append(("data_node_id", submission.data_node_id, metadata.submission.genomic_data_center_id))
+
+    # library type
+    metadata_library_type = LibraryType(get_pruefbericht_library_type(metadata))
+    if "library_type" not in ignore_fields and (submission.library_type != metadata_library_type):
+        changes.append(("library_type", submission.library_type, metadata_library_type))
+
+    # consent state
+    donor_index = next(filter(lambda d: d.relation == Relation.index_, metadata.donors))
+    consented = donor_index.consents_to_research(date=date.today())
```

Comment:
Do we only check the donor? I forgot what exactly was settled on

Reply:
You're right, we should be consistent with `grzctl consent`, meaning all donors